src/main/java/org/candlepin/model/AnonymousCloudConsumer.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) 2009 - 2023 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.model;
+
+import org.candlepin.util.Util;
+
+import org.hibernate.annotations.GenericGenerator;
+
+import java.util.Objects;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.PrePersist;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+/**
+ * An anonymous cloud consumer is an entity that does not have an owner, but is given temporary
+ * content access until it is claimed by an owner.
+ */
+@Entity
+@Table(name = AnonymousCloudConsumer.DB_TABLE)
+public class AnonymousCloudConsumer extends AbstractHibernateObject<AnonymousCloudConsumer> {
+
+    /** Name of the table backing this object in the database */
+    public static final String DB_TABLE = "cp_anonymous_cloud_consumers";
+
+    /** Max length for a value in the cloud account ID field */
+    public static final int CLOUD_ACCOUNT_ID_MAX_LENGTH = 255;
+    /** Max length for a value in the cloud instance ID field */
+    public static final int CLOUD_INSTANCE_ID_MAX_LENGTH = 85;
+    /** Max length for a value in the cloud provider short name field */
+    public static final int CLOUD_PROVIDER_SHORT_NAME_MAX_LENGTH = 15;
+    /** Max length for a value in the product ID field */
+    public static final int PRODUCT_ID_MAX_LENGTH = 32;
+
+    @Id
+    @GeneratedValue(generator = "system-uuid")
+    @GenericGenerator(name = "system-uuid", strategy = "uuid")
+    @Column(length = 32)
+    @NotNull
+    private String id;
+
+    @Column(unique = true)
+    @Size(max = 36)
```

Comment:
Size literal in the declaration.

As a side note here, these annotations trigger hard stops rather than autofill or truncate, if they even work (we've done some work to pull them out IIRC, and I can't find the annotation validator anymore); and since we use Liquibase, they're not used to build the schema, either.

if we care about these restrictions, it might make sense to enforce them at the mutator stage rather than waiting until persist.

Reply:
Updated, and moved the enforcement of the sizing to the mutators